text/0006-shipyard-next-generation.md

```diff
@@ -0,0 +1,317 @@
+
+A shipyard allows a **Site Reliability Engineer (SRE)** to specify WHAT should happen in the course of an artifact delivery or remediation process. Due to certain use cases, it has been identified that the current specification of the shipyard has limitations and hides details from the SRE. This KEP proposes an enhancement of the *Shipyard* specification to bring flexibility into the definition of the *WHAT*, but still provides an opinionated approach and smart defaults.
+
+Please read this KEP in combination with the KEP: [First version of a Keptn uniform](https://github.com/keptn/enhancement-proposals/pull/7); especially regarding the eventing part.
+
+## Explanation
+
+:man:/:blonde_woman: The target persona of this KEP is an *SRE*, who is responsible for processes and workflows to automate application delivery or remediation operations. 
+
+Before getting started, a brief recap of the important terms and definitions used in this KEP:
+
+- **Project stage:** A project stage (or just stage) defines a logical space in an environment, which has a dedicated purpose for an application in a continuous delivery process. The project stages are ordered, meaning that a stage has at least a previous and/or next stage. 
+
+- **Shipyard:** A shipyard is the declarative means to divide an environment into project stages and to specify task sequences for each stage.
+
+- **Sequence:** A sequence declares a set of tasks, which are triggered by an external event (aka. domain event, e.g.: artifact built → deploy artifact, a problem occurred → execute remediation, etc.). A sequence executes the tasks in the given order. 
+
+- **Task:** A task is the smallest executable unit of a sequence. A task is kicked-off by a *triggered* event and has a life cycle of a *start* and *finished* event. 
+
+*Current situation and Problem(s):* The current shipyard specification allows dividing an environment into project stages. In each stage, two ways for deploying (direct, blue/green) and testing (functional/performance) an artifact are provided. Besides, an SRE can specify whether a project stage is capable of handling problems by remediation actions. This specification has some limitations: 
+
+- Certain tasks are defined implicitly without giving an SRE the possibility to remove/move the task, e.g.:
+
+  - The evaluation always happens after a test execution even though the SRE already knows that no meaningful metric values are available (e.g., functional tests don't provide performance data). 
+
+  - The promotion of a deployed/released artifact happens automatically with no possibility to intervene.
+
+- It is not possible to add a task to a sequence, e.g., a task that takes care of pulling approval from a person in charge. 
+
+- The trigger (domain event) of a task sequence can not be set. 
+
+*Solution:* A revision of the shipyard specification v0.1.2 addresses the above-mentioned problems and gives SREs more flexibility in declaring processes. 
```

Comment:
> v0.1.2 
Did you mean v0.2.0

Also, I wouldn't say that "a revision of the shipyard spec" is the solution.

Maybe we should phrase it as follows:

*Solution*: By enhancing the Shipyard specificiation (v0.2.0) as detailed below, the above-mentioned problems are addressed and more flexibility for declaring processes is provided.

Reply:
I re-phrased my sentence according to your suggestion and explicitly name the target version of the new Shipyard.